src/devices/MemoryLCD/LSxxxB7DHxx.cs

```diff
@@ -0,0 +1,300 @@
+        /// </summary>
+        public int BytesPerLine { get; }
+        #endregion
+
+        #region GPIO config
+
+        /// <summary>
+        /// Chip select signal
+        /// </summary>
+        private readonly int _scs;
+
+        /// <summary>
+        /// Display ON/OFF signal<br/>The display ON/OFF signal is only for display. Data in the memory will be saved at the time of ON/OFF.<br/>When it’s "H", data in the memory will display, when it’s "L", white color will diaplay and data in the memory will be saved.
+        /// </summary>
+        private readonly int _disp;
+
+        /// <summary>
+        /// External COM inversion signal input (H: enable)<br/>When EXTMODE is "Lo", connect the EXTCOMIN to VSS.
+        /// </summary>
+        private readonly int _extcomin;
+        #endregion
+
+        #region  Delay constants for LCD timing
+
+        private static readonly int s_powerup_disp_delay = 1; // (>30us)
+        private static readonly int s_powerup_extcomin_delay = 1; // (>30us)
+        // private static readonly int s_scs_high_delay = 0; // (>3us)
+        // private static readonly int s_scs_low_delay = 0; // (>1us)
+        // private static readonly int s_interframe_delay = 0; // (>1us)
+        private static readonly int s_ts_scs = 0; // >6us
+        private static readonly int s_th_scs = 0; // >2us
+        #endregion
+
+        private GpioController _gpio;
+        private SpiDevice _spi;
+        private byte[] _lineNumberBuffer;
+        private byte[] _frameBuffer;
+
+        /// <summary>
+        /// Create a memory LCD device
+        /// </summary>
+        /// <param name="spi">SPI controller</param>
+        /// <param name="gpio">GPIO controller</param>
+        /// <param name="scs">Chip select signal</param>
+        /// <param name="disp">Display ON/OFF signal</param>
+        /// <param name="extcomin">External COM inversion signal input</param>
+        internal LSxxxB7DHxx(SpiDevice spi, GpioController gpio, int scs, int disp, int extcomin)
+        {
+            _spi = spi ?? throw new ArgumentNullException(nameof(spi));
+            _gpio = gpio;
+            _scs = scs;
+            _disp = disp;
+            _extcomin = extcomin;
+
+            BytesPerLine = (PixelWidth + 7) / 8;
+
+            _lineNumberBuffer = Enumerable.Range(0, PixelHeight).Select(m => (byte)m).ToArray();
+            _frameBuffer = new byte[BytesPerLine * PixelHeight];
+
+            Init();
+        }
+
+        /// <summary>
+        /// Updates data of only one specified line.
+        /// </summary>
+        public void DataUpdate1Line(byte lineIndex, Span<byte> data, bool frameInversion = false)
+        {
+            if (data.Length != BytesPerLine)
+            {
+                throw new Exception($"Data writing period must be {PixelWidth}ck ({BytesPerLine} bytes)");
+            }
+
+            // m(1), ag(1), d(18), dummy(2)
+            byte[] buffer = new byte[BytesPerLine + 4];
```

Comment:
Why not directly create a Span<byte> for buffer? I have hard time to understand what's the purpose of the `line` buffer. That should simplify a bit the lisibility.